allowed.py

```diff
@@ -394,56 +404,59 @@ def check_tree(tree: ast.AST, constructs: tuple, source: list) -> list:
             for alias in node.names:
                 if alias.name not in imports:
                     line = alias.lineno
                     message = f"import {alias.name}"
-                    errors.append((line, message))
+                    errors.append((get_line(line), message))
         elif isinstance(node, ast.ImportFrom):
             if node.module not in imports:
                 line = node.lineno
                 message = f"from {node.module} import ..."
-                errors.append((line, message))
+                errors.append((get_line(line), message))
             else:
                 for alias in node.names:
                     if alias.name not in imports[node.module]:
                         line = alias.lineno
                         message = f"from {node.module} import {alias.name}"
-                        errors.append((line, message))
+                        errors.append((get_line(line), message))
         elif isinstance(node, ast.Attribute) and isinstance(node.value, ast.Name):
             name = node.value
             attribute = node.attr
             if name.id in imports and attribute not in imports[name.id]:
                 line = node.lineno
                 message = f"{name.id}.{attribute}"
-                errors.append((line, message))
+                errors.append((get_line(line), message))
             elif hasattr(name, "resolved_annotation"):
                 type_name = re.match(r"[a-zA-Z.]*", name.resolved_annotation).group()
                 if type_name in methods and attribute not in methods[type_name]:
                     line = name.lineno
                     message = (
                         f"method {attribute} called on {type_name.lower()} {name.id}"
                     )
-                    errors.append((line, message))
+                    errors.append((get_line(line), message))
         elif isinstance(node, ast.Call) and isinstance(node.func, ast.Name):
             function = node.func.id
             if function in BUILTINS and function not in functions:
                 line = node.lineno
                 message = f"built-in function {function}()"
-                errors.append((line, message))
+                errors.append((get_line(line), message))
         elif isinstance(node, ast.For) and node.orelse and "for else" not in options:
             line = node.orelse[0].lineno
             message = "else in for-loop"
-            errors.append((line, message))
+            errors.append((get_line(line), message))
         elif (
             isinstance(node, ast.While) and node.orelse and "while else" not in options
         ):
             line = node.orelse[0].lineno
             message = "else in while-loop"
-            errors.append((line, message))
-    errors.sort()
+            errors.append((get_line(line), message))
+    if from_ipynb:
+        errors.sort(key=lambda e: [int(n) for n in re.findall("\d+", e[0])])
+    else:
```

Comment:
Is this needed? Since for notebooks the `line` is actually a string `"cell_C:L"` won't the normal sorting of strings sort the messages by cell and then by line? If not, this code needs a comment to explain the rationale.

Reply:
I believe this is needed, or something to deal with the numeric strings when they are not padded out with zeros. For example in python:
```python
"5" < "10" == False
"05" < "10" == True
```
I did't fancy messing about padding out single digit numbers, so I settled on this to get it working. I will see if this can be done in a less obscure way and will certainly add some clarification with comments. unless of course I am missing something obvious here which is entirely possible!

Reply:
I think you're right this is needed w/o padding. Just add a comment to explain why it's not a direct sorting of strings and don't waste time looking for a different way to implement.